osu.Desktop/Updater/AppImageUpdateManager.cs

```diff
@@ -0,0 +1,564 @@
+                /// </summary>
+                TOOL_HELP
+            }
+            /// TODO: Set commandFile to match the shipped binary to prevent searching and remove the unnecessary code
+            private string commandFile;
+            private string[] lookFor =
+            {
+                "appimageupdatetool",
+                "appimageupdatetool-x86_64.AppImage"
+            };
+            private string Command
+            {
+                get
+                {
+                    if (commandFile is default(string))
+                    {
+                        foreach (string file in lookFor)
+                        {
+                            commandFile = file;
+                            if (IsInstalled()) break;
+                        }
+                    }
+                    return commandFile;
+                }
+            }
+            private ProcessStartInfo StartInfo(ARGS args = ARGS.CHECK)
+            {
+                string arguments = "";
+                string argument_spacer = " ";
+                switch (args)
+                {
+                    case ARGS.CHECK:
+                        arguments = "--check-for-update";
+                        break;
+                    case ARGS.OVERWRITE:
+                        arguments = "--overwrite --remove-old";
+                        break;
+                    case ARGS.UPDATE:
+                        arguments = "--overwrite";
+                        break;
+                    case ARGS.TOOL_VERSION:
+                        arguments = "--version";
+                        break;
+                    case ARGS.TOOL_HELP:
+                        arguments = "--help";
+                        break;
+                    default:
+                        argument_spacer = "";
+                        break;
+                }
+
+                ProcessStartInfo info = new ProcessStartInfo
+                {
+                    FileName = Command,
+                    Arguments = $"{arguments}{argument_spacer}\"{AppImagePath}\"",
+                    UseShellExecute = false,
+                    RedirectStandardError = true,
+                    RedirectStandardOutput = true,
+                };
+
+                return info;
+            }
+            public enum States
+            {
+                NOUPDATESAVAILABLE,
+                UPDATESAVAILABLE,
+                DOWNLOADING,
+                VERIFYING,
+                COMPLETED,
+                CANCELLED
+            }
+            private static AppImageUpdateTool instance;
+            public static AppImageUpdateTool Instance
+            {
+                get
+                {
+                    if (instance is default(AppImageUpdateTool))
+                    {
+                        instance = new AppImageUpdateTool();
+                    }
+                    return instance;
+                }
+            }
```

Comment:
Is there a reason for the staticness in this class? Why not just always keep it based on instance and instantiate the tool in `AppImageUpdateManager`.

Reply:
I think the only reason for the class to be static is to check its availability( AppImageUpdateTool.IsInstalled ) before instantiating the UpdateManager so it can fallback to SimpleUpdateManager.